src/dotnet/commands/dotnet-migrate/MigrateCommand.cs

```diff
@@ -140,35 +141,35 @@ private void UpdateSolutionFile(MigrationReport migrationReport, SlnFile slnFile
                     slnPathWithTrailingSlash,
                     reportPathWithTrailingSlash);
 
-                var xprojPath = Path.Combine(relativeReportPath, report.ProjectName + ".xproj");
-                var xprojProjectsReferencedBySolution = slnFile.Projects.Where(p => p.FilePath == xprojPath);
-
                 var migratedProjectName = report.ProjectName + ".csproj";
-                if (xprojProjectsReferencedBySolution.Count() == 1)
+                var csprojPath = Path.Combine(relativeReportPath, migratedProjectName);
+                var solutionContainsCsprojPriorToMigration = slnFile.Projects
+                    .Where(p => p.FilePath == csprojPath)
+                    .Any();
+
+                if (!solutionContainsCsprojPriorToMigration)
                 {
-                    var slnProject = xprojProjectsReferencedBySolution.Single();
-                    slnProject.FilePath = Path.Combine(
-                        Path.GetDirectoryName(slnProject.FilePath),
-                        migratedProjectName);
-                    slnProject.TypeGuid = ProjectTypeGuids.CSharpProjectTypeGuid;
+                    csprojFilesToAdd.Add(Path.Combine(report.ProjectDirectory, migratedProjectName));
                 }
-                else
+
+                foreach (var preExisting in report.PreExistingCsprojDependencies)
+                {
+                    csprojFilesToAdd.Add(Path.Combine(report.ProjectDirectory, preExisting));
+                }
+
+                var projectDirectory = new DirectoryInfo(report.ProjectDirectory);
+                foreach (var xprojFileName in projectDirectory.EnumerateFiles("*.xproj"))
                 {
-                    var csprojPath = Path.Combine(relativeReportPath, migratedProjectName);
-                    var solutionContainsCsprojPriorToMigration = slnFile.Projects
-                        .Where(p => p.FilePath == csprojPath)
+                    var xprojPath = Path.Combine(relativeReportPath, xprojFileName.Name);
+                    var solutionContainsXprojFileToRemove = slnFile.Projects
+                        .Where(p => p.FilePath == xprojPath)
                         .Any();
 
-                    if (!solutionContainsCsprojPriorToMigration)
+                    if (solutionContainsXprojFileToRemove)
```

Comment:
Does "remove" fail if the sln file doesn't contain the project?  What if we just always tried to remove the project without needing to check if the project already existed in the .sln?